lib/features/family/features/admin_fee/domain/models/admin_fee.dart

```diff
@@ -21,6 +23,17 @@ class AdminFee {
     );
   }
 
+  double getTotalFee(double amount) {
```

Comment:
My domain model is kinda smart now. And alternate solution would be to create an extension on AdminFee and add these methods there. Opinions?

Reply:
Well, ultimately the nr we need to display in I is the getTotal fee so why is this not also a getter in the repo as single source of data?

Reply:
Because to me a Repository responsibility is retrieving and caching the data and being the single source of truth for the "data". But how the fee is calculated is not data but business logic ("rules"). To me it falls under the domain layer. Does that make sense?

Reply:
Yes it makes sense, then leave it here and lets have calculations like these always in the domain layer